phabfive/maniphest.py

```diff
@@ -49,115 +51,247 @@ def create_from_config(self, config_file, dry_run=False):
-            log.info(f"Title: {ticket_info['title']} | ID: {ticket_info['id']} | URI: {ticket_info['uri']}")
+                log.critical(f"Hard errors found during pre-processing step. Please update config or phabricator server and try again. Nothing has been commited yet.")
+                sys.exit(1)
+
+            # Recurse down and process all child tasks
+            processed_tasks = []
+
+            child_tasks = task_config.get("tasks", None)
+            if child_tasks:
+                for task in child_tasks:
+                    processed_tasks.append(
+                        pre_process_tasks(task)
+                    )
+
+                output["tasks"] = processed_tasks
+            else:
+                output["tasks"] = None
+
+            return output
+
+        def recurse_build_transactions(task_config):
+            """
+            This block recurses over all tasks and builds the transaction set for this ticket and stores it
+            in the data structure.
+            """
+            log.debug("Building transactions for task_config")
+            log.debug(task_config)
+
+            # In order to not cause issues with injecting data in a recurse traversal, copy the input,
+            # modify the data and return data that is later used to build a new full data structure
+            output = task_config.copy()
+
+            # Helper lambda to slim down transaction handling
+            add_transaction = lambda t, transaction_type, value : t.append({"type": transaction_type, "value": value})
+
+            transactions = []
+
+            if "title" in task_config and "description" in task_config:
+                add_transaction(transactions, "title", task_config["title"])
+                add_transaction(transactions, "description", task_config["description"])
+                add_transaction(transactions, "priority", task_config.get("priority", TICKET_PRIORITY_NORMAL))
+
+                projects = task_config.get("projects", [])
+
+                if projects:
+                    add_transaction(transactions, "projects.set", projects)
+
+                subscribers = task_config.get("subscribers", [])
+
+                if subscribers:
+                    add_transaction(transactions, "subscribers.set", subscribers)
+
+                # Prepare all parent and subtasks, and check if we have a parent task from the config file
+                subtasks = task_config.get("subtasks", [])
+
+                if subtasks:
+                    subtasks_phids = []
+
+                    for ticket_id in subtasks:
+                        search_result = self.phab.maniphest.search(
+                            constraints={"ids": [int(ticket_id)]},
+                        )
+
+                        if len(search_result["data"]) != 1:
+                            log.critical(f"Unable to find subtasks ticket with ID={ticket_id}")
+                            sys.exit(1)
```

Comment:
Use `sys.exit()` only in the CLI not in the library code. Raise an exception instead.